lms/djangoapps/ccx/views.py

```diff
@@ -210,11 +211,12 @@ def dashboard(request, course, ccx=None):
         context['is_instructor'] = ccx.is_instructor(request.user)
         context['is_staff'] = ccx.is_staff(request.user)
 
-        for ccx in custom_courses:
-            if ccx.coach.id == request.user.id:
-                context['is_ccx_coach'] = True
-                context['is_instructor'] = ccx.is_instructor(request.user)
-                context['is_staff'] = ccx.is_staff(request.user)
+        if not context['is_ccx_coach']:
```

Comment:
If context['is_ccx_coach'] is already True, we don't need to loop all custom courses.